src/test/pythonEnvironments/base/locators/composite/environmentsResolver.unit.test.ts

```diff
@@ -152,12 +160,43 @@ suite('Python envs locator - Environments Resolver', () => {
         });
     });
 
+    test('No updates except the null event is sent if environment is not safe to execute', async () => {
+        // Arrange
+        isEnvSafe.returns(false);

```

Comment:
I would move this to right above line 171.  That way it's easier to notice the relationship with `environmentSecurity`.  FWIW, it would be nice if the test used one env that is safe and one that isn't.  I'd use [sinon's `stub.callsFake()`](https://sinonjs.org/releases/v7.3.2/stubs/#stubcallsfakefakefunction).  So something like the following:

```suggestion
        isEnvSafe.callsFake((env: PythonEnvInfo) => {
            return path.basename(env.executable.filename) === 'exec1';
        });
```

Reply:
Done👍